flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/calcite/SqlRewriterUtils.scala

```diff
@@ -131,7 +131,7 @@ object SqlRewriterUtils {
     call.getKind match {
       case SqlKind.SELECT =>
         val sqlSelect = call.asInstanceOf[SqlSelect]
-        val identifiersSize = sqlSelect.getSelectList.count(s => s.isInstanceOf[SqlIdentifier])
+        val identifiersSize = sqlSelect.getSelectList.count(s => !SqlIdentifier.STAR.equals(s))
```

Comment:
Interesting.  This change makes me think that the columns could some computation.  Is it worth testing that case?